src/main/java/mate/academy/spring/config/SecurityConfig.java

```diff
@@ -1,27 +1,47 @@
 
+import mate.academy.spring.service.impl.CustomUserDetails;
 import org.springframework.beans.factory.annotation.Autowired;
-import org.springframework.context.annotation.Bean;
+import org.springframework.http.HttpMethod;
 import org.springframework.security.config.annotation.authentication.builders.AuthenticationManagerBuilder;
 import org.springframework.security.config.annotation.web.builders.HttpSecurity;
 import org.springframework.security.config.annotation.web.configuration.EnableWebSecurity;
 import org.springframework.security.config.annotation.web.configuration.WebSecurityConfigurerAdapter;
-import org.springframework.security.crypto.bcrypt.BCryptPasswordEncoder;
 import org.springframework.security.crypto.password.PasswordEncoder;
 
 @EnableWebSecurity
 public class SecurityConfig extends WebSecurityConfigurerAdapter {
+    private final PasswordEncoder passwordEncoder;
+    private final CustomUserDetails customUserDetails;
+
+    public SecurityConfig(PasswordEncoder passwordEncoder, CustomUserDetails customUserDetails) {
+        this.passwordEncoder = passwordEncoder;
+        this.customUserDetails = customUserDetails;
+    }
+
     @Autowired
     public void configure(AuthenticationManagerBuilder auth) throws Exception {
-        auth
-                .inMemoryAuthentication()
-                .passwordEncoder(getEncoder())
-                .withUser("1@123").password(getEncoder().encode("123")).roles("USER");
+        auth.userDetailsService(customUserDetails).passwordEncoder(passwordEncoder);
     }
 
     protected void configure(HttpSecurity http) throws Exception {
         http
                 .authorizeRequests()
+                .antMatchers(HttpMethod.POST,"/register/**").permitAll()
+                .antMatchers(HttpMethod.POST,"/cinema-halls/**").hasRole("ADMIN")
+                .antMatchers(HttpMethod.GET,"/cinema-halls/**").hasAnyRole("ADMIN","USER")
+                .antMatchers(HttpMethod.GET,"/movies/**").hasAnyRole("ADMIN", "USER")
+                .antMatchers(HttpMethod.POST,"/movies/**").hasRole("ADMIN")
+                .antMatchers(HttpMethod.GET,"/movies-sessions/available/**")
+                .hasAnyRole("ADMIN", "USER")
+                .antMatchers(HttpMethod.POST, "/movies-sessions/**").hasRole("ADMIN")
+                .antMatchers(HttpMethod.PUT,"/movie-session/**").hasRole("ADMIN")
+                .antMatchers(HttpMethod.DELETE,"/movies-session/**").hasRole("ADMIN")
+                .antMatchers(HttpMethod.GET,"/orders/**").hasRole("USER")
+                .antMatchers(HttpMethod.POST, "/orders/complete/**").hasRole("USER")
+                .antMatchers(HttpMethod.PUT,"/shopping-carts/movie-sessions/**").hasRole("USER")
+                .antMatchers(HttpMethod.GET,"/shopping-carts/by-user/**").hasRole("USER")
+                .antMatchers(HttpMethod.GET,"/users/by-email/**").hasRole("ADMIN")
                 .anyRequest().authenticated()
                 .and()
```

Comment:
for admin and user better create constant. private static final String ROLE_ADMIN = Role.RoleName.ADMIN.name();